src/benchmark.cc

```diff
@@ -13,36 +13,66 @@
 // limitations under the License.
 
 #include "benchmark/benchmark.h"
-#include "check.h"
-#include "colorprint.h"
-#include "commandlineflags.h"
-#include "re.h"
-#include "sleep.h"
-#include "stat.h"
-#include "sysinfo.h"
-#include "walltime.h"
 
 #include <sys/time.h>
-#include <string.h>
+#include <sys/resource.h>
+#include <unistd.h>
 
+#include <cstdlib>
+#include <cstring>
 #include <algorithm>
 #include <atomic>
 #include <condition_variable>
 #include <iostream>
 #include <memory>
-#include <mutex>
 #include <thread>
-#include <sstream>
 
-DEFINE_string(benchmark_filter, ".",
+#include "check.h"
+#include "commandlineflags.h"
+#include "colorprint.h"
+#include "log.h"
+#include "mutex.h"
+#include "re.h"
+#include "stat.h"
+#include "string_util.h"
+#include "sysinfo.h"
+#include "walltime.h"
+
+
+#if defined(__clang__)
+#
+# if __has_feature(cxx_thread_local)
+#   define BENCHMARK_THREAD_LOCAL thread_local
+# else
+#   define BENCHMARK_THREAD_LOCAL __thread
+# endif
+#
+#elif defined(__GNUC__)
+#
+# if __cplusplus >= 201103L
+#   define BENCHMARK_THREAD_LOCAL thread_local
+# else
+#   define BENCHMARK_THREAD_LOCAL __thread
+# endif
+#
+#elif defined(_MSC_VER) && !defined(__clang__)
+#
+# define BENCHMARK_THREAD_LOCAL thread_local
+#
+#endif
+
+
+DEFINE_string(benchmark_filter, "all",
               "A regular expression that specifies the set of benchmarks "
               "to execute.  If this flag is empty, no benchmarks are run.  "
               "If this flag is the string \"all\", all benchmarks linked "
               "into the process are run.");
 
-DEFINE_int32(benchmark_iterations, 0,
-             "Total number of iterations per benchmark. 0 means the benchmarks "
-             "are time-based.");
+DEFINE_int32(benchmark_min_iters, 1,
```

Comment:
I think they are useful constraints to have. Min changes the first amount of iterations attempted and max is useful for very fast benchmarks.